translator/trace/jaeger/protospan_to_jaegerproto.go

```diff
@@ -0,0 +1,535 @@
+			jTag.VStr = str
+			jTag.VType = jaeger.ValueType_STRING
+		}
+		jTags = append(jTags, jTag)
+	}
+
+	return jTags
+}
+
+func ocTimeEventsToJaegerLogsProto(ocSpanTimeEvents *tracepb.Span_TimeEvents) []jaeger.Log {
+	if ocSpanTimeEvents == nil || ocSpanTimeEvents.TimeEvent == nil {
+		return []jaeger.Log{}
+	}
+
+	ocTimeEvents := ocSpanTimeEvents.TimeEvent
+
+	// Assume that in general no time events are going to produce nil Jaeger logs.
+	jLogs := make([]jaeger.Log, 0, len(ocTimeEvents))
+	for _, ocTimeEvent := range ocTimeEvents {
+		jLog := jaeger.Log{
+			Timestamp: timestampToTimeProto(ocTimeEvent.Time),
+		}
+		switch teValue := ocTimeEvent.Value.(type) {
+		case *tracepb.Span_TimeEvent_Annotation_:
+			jLog.Fields = ocAnnotationToJagerTagsProto(teValue.Annotation)
+		case *tracepb.Span_TimeEvent_MessageEvent_:
+			jLog.Fields = ocMessageEventToJaegerTagsProto(teValue.MessageEvent)
+		default:
+			msg := "An unknown OpenCensus TimeEvent type was detected when translating to Jaeger"
+			jKV := jaeger.KeyValue{
+				Key:   ocTimeEventUnknownType,
+				VStr:  msg,
+				VType: jaeger.ValueType_STRING,
+			}
+			jLog.Fields = append(jLog.Fields, jKV)
+		}
+
+		jLogs = append(jLogs, jLog)
+	}
+
+	return jLogs
+}
+
+func ocAnnotationToJagerTagsProto(annotation *tracepb.Span_TimeEvent_Annotation) []jaeger.KeyValue {
+	if annotation == nil {
+		return []jaeger.KeyValue{}
+	}
+
+	// TODO: Find a better tag for Annotation description.
+	jKV := jaeger.KeyValue{
+		Key:   ocTimeEventAnnotationDescription,
+		VStr:  truncableStringToStr(annotation.Description),
+		VType: jaeger.ValueType_STRING,
+	}
+	return append(ocSpanAttributesToJaegerTagsProto(annotation.Attributes), jKV)
+}
+
+func ocMessageEventToJaegerTagsProto(msgEvent *tracepb.Span_TimeEvent_MessageEvent) []jaeger.KeyValue {
+	if msgEvent == nil {
+		return []jaeger.KeyValue{}
+	}
+
+	msgEventID := make([]byte, 8)
+	binary.BigEndian.PutUint64(msgEventID, uint64(msgEvent.Id))
+
+	uncompressedSize := make([]byte, 8)
+	binary.BigEndian.PutUint64(uncompressedSize, uint64(msgEvent.UncompressedSize))
+
+	compressedSize := make([]byte, 8)
+	binary.BigEndian.PutUint64(compressedSize, uint64(msgEvent.CompressedSize))
+
+	// TODO: Find a better tag for Message event.
+	msgTypeStr := msgEvent.Type.String()
+	jaegerKVs := []jaeger.KeyValue{
+		jaeger.KeyValue{
+			Key:   ocTimeEventMessageEventType,
+			VStr:  msgTypeStr,
+			VType: jaeger.ValueType_STRING,
+		},
+		jaeger.KeyValue{
+			Key:     ocTimeEventMessageEventID,
+			VBinary: msgEventID,
+			VType:   jaeger.ValueType_BINARY,
+		},
+		jaeger.KeyValue{
+			Key:     ocTimeEventMessageEventUSize,
+			VBinary: uncompressedSize,
+			VType:   jaeger.ValueType_BINARY,
+		},
+		jaeger.KeyValue{
+			Key:     ocTimeEventMessageEventCSize,
+			VBinary: compressedSize,
+			VType:   jaeger.ValueType_BINARY,
+		},
+	}
+	return jaegerKVs
+}
+
+func traceIDBytesToUnsignedLowAndHigh(traceID []byte) (traceIDLow, traceIDHigh uint64, err error) {
```

Comment:
nit: can we move this into trace/translator/big_endian_converter.go potentially? It seems generically useful

Reply:
The functions handle special case error reporting, referencing errors defined in 
trace/translator/jaeger/constants.go as of now
Would you recommend we remove those references and put the functions in 
big_endian_converter.go?

Reply:
yea I think those errors should also be moved up. I think you use these errors elsewhere, but only as a guard against the same things that are checked here (nil and wrong length). I think it should be fine to remove the other guards, let this function do the safety checking, and then remove the other references to these errors. What are your thoughts about that?

Reply:
You're right. I'll make the changes.